src/vsphere_cpi/lib/cloud/vsphere/device_key_gen.rb

```diff
@@ -0,0 +1,15 @@
+module VSphereCloud
+  module DeviceKeyGenerator
+    def self.init(keys: nil)
+      @memo = Set.new(keys)
+      @seed = -1
+    end
+
+    def self.get_device_key
+      until @memo.add?(@seed) != nil
```

Comment:
Probably should be `while @memo.add?(@seed).nil?` to use `#nil?` rather than `== nil`